### ISSUE TITLE
Shared build not working?

### ISSUE THREAD

**Issue by baentsch** [google/cpu_features#108]
Positive statement first: All works fine when doing static builds. Our project that wants to use `cpu_features` has to build its libraries shared, though, too, and there I'm running into problems:

I read and followed these suggestions (https://github.com/google/cpu_features/issues/44 ) as well as the [subproject build instructions](https://github.com/google/cpu_features/blob/master/cmake/README.md) but to no avail: 
1) When doing a `cpu_features` "local" `cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=ON -DBUILD_PIC=ON -H. -Bcmake_build` this is the result of `make` on Ubuntu:
```[ 77%] Linking C shared library libcpu_features.so
/usr/bin/ld: CMakeFiles/utils.dir/src/string_view.c.o: relocation R_X86_64_PC32 against symbol `CpuFeatures_StringView_IndexOf' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: Bad value
collect2: error: ld returned 1 exit status
CMakeFiles/cpu_features.dir/build.make:100: recipe for target 'libcpu_features.so' failed
make[2]: *** [libcpu_features.so] Error 1
CMakeFiles/Makefile2:178: recipe for target 'CMakeFiles/cpu_features.dir/all' failed
make[1]: *** [CMakeFiles/cpu_features.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2
```

2) When doing this embedded in our larger project, not only can I not build the shared lib, even this happens during `make`:
```
[ 95%] Linking C executable list_cpu_features
CMakeFiles/list_cpu_features.dir/src/utils/list_cpu_features.c.o: In function `CreateTree':
list_cpu_features.c:(.text+0x575): undefined reference to `GetX86Info'
list_cpu_features.c:(.text+0x57d): undefined reference to `GetX86CacheInfo'
list_cpu_features.c:(.text+0x585): undefined reference to `FillX86BrandString'
list_cpu_features.c:(.text+0x676): undefined reference to `GetX86Microarchitecture'
list_cpu_features.c:(.text+0x67d): undefined reference to `GetX86MicroarchitectureName'
list_cpu_features.c:(.text+0x6c9): undefined reference to `GetX86FeaturesEnumValue'
list_cpu_features.c:(.text+0x6d9): undefined reference to `GetX86FeaturesEnumName'
collect2: error: ld returned 1 exit status
src/cpu_features/CMakeFiles/list_cpu_features.dir/build.make:95: recipe for target 'src/cpu_features/list_cpu_features' failed
make[2]: *** [src/cpu_features/list_cpu_features] Error 1
CMakeFiles/Makefile2:432: recipe for target 'src/cpu_features/CMakeFiles/list_cpu_features.dir/all' failed
make[1]: *** [src/cpu_features/CMakeFiles/list_cpu_features.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
```

--> Key question: Is the use of `cpu_features` as a shared lib not just discouraged, but even no longer supported? If so do you see any way how to integrate it into a project that _has_ to build its libraries shared? If so could you update your [guidance for integration](https://github.com/google/cpu_features/blob/master/cmake/README.md) ? Thanks in advance!

**Comment by Mizux** [comment]
According to the code:
https://github.com/google/cpu_features/blob/b5c271c53759b2b15ff91df19bd0b32f2966e275/CMakeLists.txt#L17-L24

So i would say it is discouraged but it should be supported. We currently don't have any CI jobs testing it since we don't have usage of this but i'll try to fix it ASAP (at least on linux since symbols are not hidden and you don't need dllimport/dllexport shenanigan like on windows...)

**Comment by baentsch** [comment]
@Mizux Thanks for the quick reaction. I also agree that activating PIC automatically when BUILD_SHARED_LIBS is ON makes sense: I added it manually in my test above because I saw the explicit OFF in the baseline CMakeLists--but that didn't change the outcome... 
FYI, while [our project](https://github.com/open-quantum-safe) has a cross-platform (incl. Windows) goal we'd need `cpu_features` primarily for Linux Docker images running (without missing CPU-features induced crashes :) on different CPUs, so seeing this work for Linux would be perfect/sufficient right now. So Thanks again for taking this on!
